compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2119,8 +2330,10 @@ private void validateReadOnlyIntersectionTypeDefinitions(List<BLangTypeDefinitio
                     if (constituentType == symTable.readonlyType) {
                         continue;
                     }
+                    // If constituent type is error, we have already validated error intersections.
+                    if (!types.isSelectivelyImmutableType(constituentType, true, true)
+                            && constituentType.tag != TypeTags.ERROR) {
```

Comment:
Isn't adding this check a bit contradictory with the method name? Should we rename the method to be more generic?